### PR TITLE
Fixing a bug when MlFlow try to log a torch.tensor

### PR DESCRIPTION
**Pull request by etiennebonnafoux**
# What does this PR do?

Add the case where a torch.tensor with one element is log on Mlflow. For now it only accept float and int.

Fixes # (issue)

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [X] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#create-a-pull-request),
      Pull Request section?
- [X] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
Here https://github.com/huggingface/transformers/issues/29784
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
No need
- [ ] Did you write any new necessary tests?
No need (two lines of code)

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

@amyeroberts

### CODE REVIEW

File: src/transformers/integrations/integration_utils.py [hunk 1]
```diff
@@ -1071,6 +1071,8 @@ def on_log(self, args, state, control, logs, model=None, **kwargs):
             for k, v in logs.items():
                 if isinstance(v, (int, float)):
                     metrics[k] = v
+                elif isinstance(v, torch.Tensor) and v.numel() == 1:
+                   metrics[k] = float(v)
```
Comment (amyeroberts) [inline src/transformers/integrations/integration_utils.py:1075]: Let's use `.item()` here so we preserve if it's a float or int 

```suggestion
                   metrics[k] = v.item()
```

**Review by amyeroberts — APPROVED** [review]
Thanks for fixing! Just a small suggestion

**Comment by etiennebonnafoux** [comment]
Sorry but what should I do to pass  check_code_quality  and setup_and_quality  test ? this is my first PR

**Comment by amyeroberts** [comment]
@etiennebonnafoux Running `make fixup` and pushing the changes should resolve this. There's more info in the [contributing guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#create-a-pull-request)

**Pull request merged**